exp/views/study.py

```diff
@@ -873,9 +877,60 @@ def post(self, request, *args, **kwargs):
         TODO: No more POST masquerading as GET, rewrite the template and both the
             web and researcher views to be more reasonably sane.
         """
-        child_id = self.request.POST.get("child_id")
-        kwargs["child_id"] = child_id
-        return HttpResponseRedirect(reverse("exp:preview-proxy", kwargs=kwargs))
+        study: Study = self.get_object()
+        child: Child = Child.objects.get(uuid=self.request.POST.get("child_id"))
+
+        if study.study_type.is_ember_frame_player:
+            # EFP confuses "id" and "uuid"
+            kwargs["child_id"] = child.uuid
+            return HttpResponseRedirect(reverse("exp:preview-proxy", kwargs=kwargs))
+        else:
+            return redirect(
+                reverse(
+                    "exp:preview-jspsych", kwargs={"pk": study.id, "child_id": child.id}
+                )
+            )
```

Comment:
Ah, this might answer my question about why external study previews are redirecting to a jsPsych URL. It looks like we need an `if study.study_type.is_external: ...`?

Reply:
This is preview detail view.  There hasn't been a need for a redirect to external preview.